dom.go

```diff
@@ -2,9 +2,65 @@ package vecty
 
 import "github.com/gopherjs/gopherjs/js"
 
+// Renderable is a user-defined renderable component.
+type Renderable interface {
+	Component
+	Render() Component
+}
+
+// Core is the struct which all components embed.
+type Core struct {
+	component Renderable
+	body      Component
+}
+
+// Apply implements the Markup interface.
+func (c *Core) Apply(e *Element) {
+	e.AddChild(c.component)
+}
+
+// Unmount implements the Component interface.
+func (c *Core) Unmount() {
+	// Break the circular link between c (Core) and c.component (Renderable) or
+	// else one object could not be garbage collector without the other (both
+	// would have to be unused).
+	c.component = nil
+	c.body.Unmount()
+}
+
+// Node implements the Component interface.
+func (c *Core) Node() *js.Object {
+	return c.body.Node()
+}
+
+// Reconcile implements the Component interface.
+func (c *Core) Reconcile(oldComp Component) {
```

Comment:
Why is there `oldComp Component` here and not used? This looks like the old `ReconcileBody` function, but now it is mixed with the "reconcile component" part. This looks strange to me, since those are two distinct steps:
1. You ask a sub-component to reconcile with the previous instance of the sub-component.
2. The component does that and then reconciles its body by rendering and then asking the body to reconcile with the old body (step 1 again).